### ISSUE TITLE
Help Understanding / Confusing project references

### ISSUE THREAD

**Issue by groksrc** [brianegan/flutter_architecture_samples#101]
Hello, I really appreciate this project. It's helped tremendously in my understanding of Flutter. That said, I had a fair amount of trouble getting my head wrapped around it because not only am I new to Flutter but the organization of the project really made me stretch as I was figuring out how things fit together. I'd like to propose a few changes to make it easier for others. I've already made most of these in my [fork](https://github.com/groksrc/flutter_architecture_samples) but I'm happy to open a PR if it's not too invasive. Here are my proposed changes:

* All folder names under the `example` directory should match the name of the library in the pubspec.yml
* All libraries in the `example` directory that are 'front-end' libraries (those that expose a UI via a main.dart) should start with `flutter_` and be named with the pattern `flutter_<implementation>`.
* Supporting libraries should start with the same name as the UI package unless they are shared across multiple projects, in which case they should start with `todos_`
* Front-end packages should not take a reference on other front end packages
* Project exports should all have a main export file. For front-end libraries that file is main.dart. For the rest it is <library_name>.dart. 
* Project imports from the same library should use a relative reference so that it is clear the import originates from the package you are already in. The Dart style guide regarding package imports should be followed.
* The projects that handle repository behavior should begin with `todos_repository_`

When you do these things, you wind up with an example directory that looks like this:
![screen shot 2019-01-31 at 10 18 22 am](https://user-images.githubusercontent.com/9099311/52068296-0268c280-2542-11e9-8a7c-6d15b46d424c.png)

**Comment by groksrc** [comment]
Great! Since we're moving out of the `example` directory I should be able to do clean PRs into the new location. By convention the projects would move to `lib/src` so combining these ideas here's how I think it would shape up in terms of PRs:

1) Consolidate what is currently in `lib/src` into a `todos_app_base` library. (As a side benefit we should be able to remove the duplicate uuid and localization files later on with more ease.)
2) Move all of the non bloc projects to `lib/src`
3) Move the bloc projects. I'm a little fuzzy on this one so let me know.

If I have it right the directory structure would ultimately look like this:
```
* lib/src/bloc
* lib/src/bloc/base
* lib/src/bloc/base/standard ?  ( formerly example/blocs )
* lib/src/bloc/base/simple      ( formerly example/simple_blocs )
* lib/src/bloc/mobile
* lib/src/bloc/web
* lib/src/flutter_built_redux
* lib/src/flutter_firestore_redux
* lib/src/flutter_inherited_widget
* lib/src/flutter_mvc
* lib/src/flutter_mvi
* lib/src/flutter_mvi_base
* lib/src/flutter_mvu
* lib/src/flutter_redurx
* lib/src/flutter_redux
* lib/src/flutter_scoped_model
* lib/src/flutter_vanilla
* lib/src/integration_tests
* lib/src/todos_app_base
* lib/src/todos_repository_base
* lib/src/todos_repository_firebase
* lib/src/todos_repository_firebase_rtdb
* lib/src/todos_repository_simple
```
Since this is a reasonably large change it might make sense to start a new branch off of master so that I can PR against that instead and once everything is settle you could merge it up? I'll PR against master to begin with and look for feedback.

Thanks again!

**Comment by brianegan** [comment]
> Move all of the non bloc projects to lib/src

I was thinking of moving everything to the root directory for two reasons: First, I figured that would make them easier to find. Second, like the `example` directory, I think putting the code in `lib/src` would make more sense if all of those belonged to the same app or project, but they're all independent pieces.

Thoughts?

  > Move the bloc projects. I'm a little fuzzy on this one so let me know.

Let's keep `bloc` and `simple_bloc` separate for now, since they're meant to be two separate projects! So we'd have:

```
/
  -vanilla
  -scoped_model
  -bloc
    -core
    -mobile
    -web
  -simple_bloc
    -core
    -mobile
    -web
```

  > Since this is a reasonably large change it might make sense to start a new branch off of master so that I can PR against that instead and once everything is settled you could merge it up?

Yep, sounds great! Thanks again for all the help and raising the issue :)

**Comment by groksrc** [comment]
I agree that `lib/src` is probably not ideal since you're only supposed to put private library code in there, but putting each sample into the project root doesn't feel right either. According to the [docs](https://www.dartlang.org/guides/libraries/create-library-packages#what-makes-a-library-package) it looks like putting them in `lib` may be the right choice since that folder is intended to be public? So then we would have:
```
/
  -lib
    -bloc
      -core
      -mobile
      -web
    -flutter_scoped_model
    -flutter_vanilla
    -flutter_<implementation>
    -simple_bloc
      -core
      -mobile
      -web
    -todos_app_base
    -todos_repository_base
    -todos_repository_<implementation>
```
Does that work?

I noticed we're using the terms `core` and `base` for sort-of the same thing. We probably want to pick one or the other and stick with it for consistency, I'm indifferent as to which so lmk if you have a preference.

I enjoy the collaborative PR approach so I'll send you my first draft pretty quick.

**Comment by brianegan** [comment]
Yep -- definitely put stuff inside the `lib` directory when it comes to packages! 

However, in my head, I don't think of the root of this project as a Dart package (it is currently, but that's a bit of a mistake :P). I see the root as a place that holds some Dart packages (some of them are Flutter apps, others provide supporting code), and each of those packages have their own `lib` directory.

You can see a similar approach on official Dart projects like json_serializable, where the Repo contains a few Dart packages at the root of the project, and each package has the `lib` directory within.

  > I noticed we're using the terms core and base for sort-of the same thing.

Good call. Let's go with `core` -- I've seen it used more commonly in other examples.

**Comment by groksrc** [comment]
That's reasonable, I'll go ahead and update my first PR to reflect this feedback. 👍

**Comment by mmcc007** [comment]
Closed by #104, #110, #111, #112, #118
